homeassistant/components/netgear/device_tracker.py

```diff
@@ -1,166 +1,141 @@
+_LOGGER = logging.getLogger(__name__)
 
-        self.last_results = results or []
 
-    def get_attached_devices(self):
-        """List attached devices with pynetgear.
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up device tracker for Netgear component."""
+    router = hass.data[DOMAIN][entry.unique_id]
+    tracked = set()
+
+    @callback
+    def update_router():
+        """Update the values of the router."""
+        add_entities(router, async_add_entities, tracked)
+
+    router.listeners.append(
+        async_dispatcher_connect(hass, router.signal_device_new, update_router)
+    )
+
+    update_router()
+
+
+@callback
+def add_entities(router, async_add_entities, tracked):
+    """Add new tracker entities from the router."""
+    new_tracked = []
+
+    for mac, device in router.devices.items():
+        if mac in tracked:
+            continue
+
+        new_tracked.append(NetgearDevice(router, device))
+        tracked.add(mac)
+
+    if new_tracked:
+        async_add_entities(new_tracked, True)
+
+
+class NetgearDevice(ScannerEntity):
+    """Representation of a Netgear device."""
+
+    def __init__(self, router: NetgearRouter, device: Dict[str, any]) -> None:
+        """Initialize a Netgear device."""
+        self._router = router
+        self._name = device["name"]
+        self._mac = device["mac"]
+        self._manufacturer = device["device_model"]
+        self._icon = icon_for_device(device)
+        self._active = True
+        self._attrs = {}
+
+    def update(self) -> None:
+        """Update the Netgear device."""
+        device = self._router.devices[self._mac]
+        self._active = device["active"]
+        self._attrs = {
+            "link_type": device["type"],
+            "link_rate": device["link_rate"],
+            "signal_strength": device["signal"],
+        }
+        if not self._active:
+            self._attrs = {}
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return self._mac
+
+    @property
+    def name(self) -> str:
+        """Return the name."""
+        return self._name
+
+    @property
+    def is_connected(self):
+        """Return true if the device is connected to the network."""
+        return self._active
+
+    @property
+    def source_type(self) -> str:
+        """Return the source type."""
+        return SOURCE_TYPE_ROUTER
+
+    @property
+    def icon(self) -> str:
+        """Return the icon."""
+        return self._icon
+
+    @property
+    def device_state_attributes(self) -> Dict[str, any]:
+        """Return the attributes."""
+        return self._attrs
+
+    @property
+    def device_info(self) -> Dict[str, any]:
+        """Return the device information."""
+        return {
+            "connections": {(CONNECTION_NETWORK_MAC, self._mac)},
+            "identifiers": {(DOMAIN, self.unique_id)},
```

Comment:
You can leave this out, just a mac is enough.

Reply:
What do you mean ?

Just replacing `self.unique_id` by `self._mac` ?
I actually "like" that, now they are the same but might be different, so I can reuse the code :grin:

Reply:
We match devices on connections + identifiers. They are treated the same. There is no reason to specify it twice.